eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/ReplaceableTx.scala

```diff
@@ -0,0 +1,200 @@
+
+  override def sign(extraUtxos: Map[OutPoint, TxOut]): ReplaceableRemoteCommitAnchor = {
+    copy(txInfo = txInfo.sign(fundingKey, commitKeys, commitment.params.commitmentFormat, extraUtxos))
+  }
+}
+
+/**
+ * A transaction spending an HTLC output from the local commitment.
+ * We always keep the HTLC input and output at index 0 for simplicity.
+ */
+sealed trait ReplaceableHtlc extends ReplaceableTxWithWalletInputs {
+  override def reset(): ReplaceableHtlc = {
+    val updatedTx = txInfo.tx.copy(txIn = txInfo.tx.txIn.take(1), txOut = txInfo.tx.txOut.take(1))
+    updateTx(updatedTx)
+  }
+
+  def setChangeOutput(amount: Satoshi, changeScript: ByteVector): ReplaceableHtlc = {
+    // We always keep the HTLC output in the first position.
+    val updatedTx = txInfo.tx.copy(txOut = txInfo.tx.txOut.take(1) ++ Seq(TxOut(amount, changeScript)))
+    updateTx(updatedTx)
+  }
+
+  def updateChangeAmount(amount: Satoshi): ReplaceableHtlc = {
+    if (txInfo.tx.txOut.size > 1) {
+      setChangeOutput(amount, txInfo.tx.txOut.last.publicKeyScript)
+    } else {
+      this
+    }
+  }
+
+  def removeChangeOutput(): ReplaceableHtlc = {
+    val updatedTx = txInfo.tx.copy(txOut = txInfo.tx.txOut.take(1))
+    updateTx(updatedTx)
+  }
+
+  override protected def updateTx(tx: Transaction): ReplaceableHtlc = this match {
+    case htlcTx: ReplaceableHtlcSuccess => htlcTx.copy(txInfo = htlcTx.txInfo.copy(tx = tx))
+    case htlcTx: ReplaceableHtlcTimeout => htlcTx.copy(txInfo = htlcTx.txInfo.copy(tx = tx))
+  }
+}
+
+case class ReplaceableHtlcSuccess(txInfo: HtlcSuccessTx, commitKeys: LocalCommitmentKeys, preimage: ByteVector32, remoteSig: ByteVector64, commitTx: Transaction, commitment: FullCommitment) extends ReplaceableHtlc {
+  override def redeemInfo(): RedeemInfo = txInfo.redeemInfo(commitKeys.publicKeys, commitment.params.commitmentFormat)
+
+  override def sign(extraUtxos: Map[OutPoint, TxOut]): ReplaceableHtlcSuccess = {
+    val localSig = txInfo.sign(commitKeys, commitment.params.commitmentFormat, extraUtxos)
+    copy(txInfo = txInfo.addSigs(commitKeys, localSig, remoteSig, preimage, commitment.params.commitmentFormat))
+  }
+}
+
+case class ReplaceableHtlcTimeout(txInfo: HtlcTimeoutTx, commitKeys: LocalCommitmentKeys, remoteSig: ByteVector64, commitTx: Transaction, commitment: FullCommitment) extends ReplaceableHtlc {
+  override def redeemInfo(): RedeemInfo = txInfo.redeemInfo(commitKeys.publicKeys, commitment.params.commitmentFormat)
+
+  override def sign(extraUtxos: Map[OutPoint, TxOut]): ReplaceableHtlcTimeout = {
+    val localSig = txInfo.sign(commitKeys, commitment.params.commitmentFormat, extraUtxos)
+    copy(txInfo = txInfo.addSigs(commitKeys, localSig, remoteSig, commitment.params.commitmentFormat))
+  }
+}
+
+/** A transaction spending an HTLC output from a remote commitment. */
+sealed trait ReplaceableClaimHtlc extends ReplaceableTx {
```

Comment:
We could add a comment to explain that this type is different from `ReplaceableHtlc` because we won't add wallet inputs, we'll update the output amount instead ? It may not be 100% necessary since you could argue that the types are slef-explanatory